electron_app/electron.js

```diff
@@ -324,7 +326,25 @@ Main._createMainWindow = function () {
   // history.
   Main._window.loadURL('/');
 
-  Main._window.on('closed', () => {
+  Main._window.on('close', (event) => {
+    if (canNotCloseApplication) {
+      event.preventDefault();
+
+      Main._window.webContents.send('show-close-modal');
+
+      return false;
+    }
+  });
+
+  electron.ipcMain.on('close-bpmn-studio', (event) => {
+    Main._window.close();
+  });
+
+  electron.ipcMain.on('can-not-close', (event, data) => {
```

Comment:
I'd rename the second parameter of the lambda function to something more descriptive then just `data`.

```suggestion
  electron.ipcMain.on('can-not-close', (event, canCloseResult) => {
```